src/twig.logic.js

```diff
@@ -616,10 +615,7 @@ module.exports = function (Twig) {
                 }
 
                 // Render the template in case it puts anything in its context
-                template.render(innerContext);
-
-                // Extend the parent context with the extended context
-                Twig.lib.extend(context, innerContext);
```

Comment:
Since the context object is passed by reference (instead of a copy), any changes made to the context when rendered are made on the same instance. This made the call to `Twig.lib.extend()` extraneous, so I removed it. #369 added the `template.render()` call which applies any changes to the context and is retained below.